plover/gui_qt/dictionaries_widget.py

```diff
@@ -327,6 +381,59 @@ def on_edit_dictionaries(self):
         assert selection
         self._edit([self._config_dictionaries[row] for row in selection])
 
+    def _copy_dictionaries(self, dictionaries_list):
+        need_reload = False
+        title_template = _('Save a copy of {name} as...')
+        default_name_template = _('{name} - Copy')
+        for dictionary in dictionaries_list:
+            title = title_template.format(name=dictionary.short_path)
+            name, ext = os.path.splitext(os.path.basename(dictionary.path))
+            default_name = default_name_template.format(name=name)
+            new_filename = _get_dictionary_save_name(self, title, default_name, [ext[1:]],
+                                                     initial_filename=dictionary.path)
+            if new_filename is None:
+                continue
+            with _new_dictionary(new_filename) as d:
+                d.update(self._loaded_dictionaries[dictionary.path])
+            need_reload = True
```

Comment:
Just tried using this feature, and wondering if it makes sense when handling multiple dictionaries to remember the path that it was saved to.

I.e. instead of dictionary.path for initial_filename, do that for only the first, and then on subsequent dialogs open with where `new_filename` was pointing to.

I could definitely see wanting to make a copy of X, Y, Z dictionaries and hoping that they all end up in the same folder.

(I accidentally didn't realize that the path wasn't sticky and just tried to blindly "save" and it saved to the wrong folder)